packages/sanity/src/adapter.ts

```diff
@@ -0,0 +1,159 @@
+import type { Adapter } from "next-auth/adapters"
+import type { Session, Profile, User } from "next-auth"
+import {
+  getUserByIdQuery,
+  getUserByProviderAccountIdQuery,
+  getUserByEmailQuery,
+} from "./queries"
+import LRU from "lru-cache"
+import { SanityClient } from "@sanity/client"
+
+interface Options {
+  client: SanityClient
+}
+
+const userCache = new LRU<string, User & { id: string }>({
+  maxAge: 24 * 60 * 60 * 1000,
+  max: 1000,
+})
+
+export const SanityAdapter: Adapter<
+  Options,
```

Comment:
So for a unified experience across the adapters, I suggest that the first argument of the `Adapter` is always simply the client that is used to talk to the database, and any additional options are passed as the second argument if needed.

```suggestion

const userCache = new LRU<string, User & { id: string }>({
  maxAge: 24 * 60 * 60 * 1000,
  max: 1000,
})

export const SanityAdapter: Adapter<
  SanityClient,
```